internal/view/owner_extender.go

```diff
@@ -0,0 +1,154 @@
+package view
+
+import (
+	"errors"
+	"fmt"
+	"github.com/derailed/k9s/internal/dao"
+	"github.com/derailed/k9s/internal/ui"
+	"github.com/derailed/tcell/v2"
+	"github.com/derailed/tview"
+)
+
+const selectOwnerDialogKey = "owner"
+
+// OwnerExtender adds log actions to a given viewer.
+type OwnerExtender struct {
+	ResourceViewer
+}
+
+// NewOwnerExtender returns a new extender.
+func NewOwnerExtender(v ResourceViewer) ResourceViewer {
+	o := OwnerExtender{
+		ResourceViewer: v,
+	}
+	o.AddBindKeysFn(o.bindKeys)
+
+	return &o
+}
+
+// BindKeys injects new menu actions.
+func (o *OwnerExtender) bindKeys(aa ui.KeyActions) {
+	res, err := dao.AccessorFor(o.App().factory, o.GVR())
```

Comment:
Don't need to pre-check here - if the extension is used the view will surface the action.